phases/19-capstone-projects/81-end-to-end-distributed-train/code/main.py

```
@@ -0,0 +1,444 @@
+        os.fsync(f.fileno())
+    for tmp, final in tmps:
+        os.replace(tmp, final)
+    os.replace(manifest_tmp, dest / "manifest.json")
+    return manifest
+
+
+def load_sharded(src_dir: str, expected_world_size: int) -> list:
+    src = Path(src_dir)
+    manifest = json.loads((src / "manifest.json").read_text())
+    if manifest["world_size"] != expected_world_size:
+        raise RuntimeError(f"world_size mismatch")
```

Ruff (0.15.14)

[error] f-string without any placeholders

Remove extraneous `f` prefix

(F541)